File: backend_java_springboot/src/main/java/top/laonai/texttospeechbackend/module/UserInfo.java
package top.laonai.texttospeechbackend.module;


import jakarta.persistence.*;
import org.springframework.security.crypto.bcrypt.BCrypt;

@Entity
@Table(name = "user_info")
public class UserInfo {

@Id
@GeneratedValue(strategy = GenerationType.IDENTITY)
private Long id;

@Column(name = "username", nullable = false, unique = true)
private String username;

@Column(name = "password", nullable = false)
private String password;

// 设置密码时进行哈希
public void setPassword(String password) {
this.password = BCrypt.hashpw(password, BCrypt.gensalt());
}

// 检查密码是否正确
public boolean checkPassword(String plainPassword) {
return BCrypt.checkpw(plainPassword, this.password);
}

// Getters 和 Setters
public Long getId() {
return id;
}

public void setId(Long id) {
this.id = id;
}

public String getUsername() {
return username;
}

public void setUsername(String username) {
this.username = username;
}

public String getPassword() {
return password;
}
}

File: backend_flask/module/user_info.py
from app import db
from werkzeug.security import generate_password_hash, check_password_hash

class user_info(db.Model):
    __tablename__ = 'user_info'
    id = db.Column(db.Integer, primary_key=True)
    username = db.Column(db.String(80), unique=True, nullable=False)
    password = db.Column(db.String(128), nullable=False)  # 保存密码哈希值

    # 设置密码时，保存哈希
    def set_password(self, password):
        self.password = generate_password_hash(password)

    # 验证密码时，进行哈希比对
    def check_password(self, password):
        return check_password_hash(self.password, password)  # 对比哈希值
